l10n_es_aeat_silicie/models/stock_move.py

```diff
@@ -0,0 +1,508 @@
+                        move.factor_conversion_silicie = move.product_id.product_tmpl_id.factor_conversion_silicie
+                        move.alcoholic_grade = move.product_id.product_tmpl_id.alcoholic_grade
+                        move.nc_code = move.product_id.product_tmpl_id.nc_code
+                        move.product_key_silicie_id = move.product_id.product_tmpl_id.product_key_silicie_id
+                        move.container_type_silicie_id = move.product_id.product_tmpl_id.container_type_silicie_id
+                        move.epigraph_silicie_id = move.product_id.product_tmpl_id.epigraph_silicie_id
+                        move.uom_silicie_id = move.product_id.product_tmpl_id.uom_silicie_id
+                        move.fiscal_position_id = move.picking_id.partner_id.property_account_position_id
+                        # Check if all fields have been correctly generated
+                        move.check_silicie_fields()
+            '''
+            else:
+                move.clear_silicie_fields()
+            '''
+
+    @api.multi
+    def clear_silicie_fields(self):
+        for move in self:
+            move.silicie_product_type = 'none'
+            move.factor_conversion_silicie = 0
+            move.alcoholic_grade = 0
+            move.nc_code = ''
+            move.product_key_silicie_id = False
+            move.container_type_silicie_id = False
+            move.epigraph_silicie_id = False
+            move.uom_silicie_id = False
+            move.fiscal_position_id = False
+            move.silicie_proof_type_id = False
+            move.silicie_move_type_id = False
+            move.silicie_processing_id = False
+            move.silicie_loss_id = False
+            move.notes_silice = ''
+
+    def _action_done(self):
+        move = super(StockMove, self)._action_done()
+        move.generate_silicie_fields()
+        return move
+
+    @api.model
+    def cron_generate_silicie_fields(self):
+        moves = self.env['stock.move'].search(
+            [("state", "=", 'done'),
+             ('silicie_product_type', '!=', 'none'),
+             ('date_send_silicie', '=', False)])
+        moves.generate_silicie_fields()
+        moves = self.env['stock.move'].search(
+            [("state", "=", 'done'),
+             ('silicie_product_type', '!=', 'none')])
+        moves.check_silicie_fields()
+        return
+
+    @api.multi
+    def check_silicie_fields(self):
+        a14_type = self.env.ref(
+            'l10n_es_aeat_silicie.aeat_move_type_silicie_a14')
+        a15_type = self.env.ref(
+            'l10n_es_aeat_silicie.aeat_move_type_silicie_a15')
+        a32_type = self.env.ref(
+            'l10n_es_aeat_silicie.aeat_move_type_silicie_a32')
+        silicie_domain = ('state', '=', 'done')
+        for record in self:
+            if record.silice_tax_position and record.silicie_proof_type_id and record.epigraph_silicie_id \
```

Comment:
Este if lo simplificaría. Haría un método que devolviera una lista de nombres de fields a comprobar (la llamada fuera del for), que además sería fácil de heredar, ejemplo:
```
@api.model
def get_fields_to_check(self):
    return [
        'silice_tax_position', 'silicie_proof_type_id', 'epigraph_silicie_id', 
        'product_key_silicie_id', 'container_type_silicie_id', 'uom_silicie_id', 
        'qty_conversion_silicie', 'alcoholic_grade', 'absolute_alcohol'
    ]
```
y luego el if algo así:
`if all(getattr(record, field_to_check, False) for field_to_check in fields_to_check)`